### ISSUE TITLE
What role does gorm play in managing connection pool? 

### ISSUE THREAD

**Issue by Xieyuschen** [go-gorm/gorm#3588]
I'm new to `gorm` and just try some samples in docs. When I want to use gorm in a goroutine I encountered  `Error 1040: Too many connections` when I execute codes as follows:
```go
func main() {
        //util.GetTimeStamp() generate a time stamp with int64 type
        //I use it here to show how long this block of codes spend
	before := util.GetTimeStamp()
	for i:=0;i<1000;i++{
                //there are 10k elements in table and QueryFromDbRandomly get one of them randomly
		go services.QueryFromDbRandomly()
	}
	now := util.GetTimeStamp()
	log.Println(now-before)
	var t string
	fmt.Scanln(&t)
}
```
I encoutered `Error 1040: Too many connections` when execute the codes. I don't know whether it's caused because too many goroutine want to use connections which is managed by gorm. Or the error just comes from mysql driver without any management of gorm.
In short, what does gorm play in managing connection pool ? And does gorm do something like a connection pool drive:)

**Comment by jinzhu** [comment]
https://gorm.io/docs/connecting_to_the_database.html#Connection-Pool
